src/basic/path-util.c

```diff
@@ -851,19 +853,21 @@ int path_extract_filename(const char *p, char **ret) {
         const char *c, *e = NULL, *q;
 
         /* Extracts the filename part (i.e. right-most component) from a path, i.e. string that passes
-         * filename_is_valid(). A wrapper around last_path_component(), but eats up trailing slashes. */
+         * filename_is_valid(). A wrapper around last_path_component(), but eats up trailing slashes. Returns
+         * -EADDRNOTAVAIL if specified parameter includes no filename (i.e. is "/" or so). Returns -EINVAL if
+         * not a valid path in the first place. */
 
-        if (!p)
+        if (!path_is_valid(p))
                 return -EINVAL;
 
         c = last_path_component(p);
 
-        for (q = c; *q != 0; q++)
+        for (q = c; *q != 0; q++) /* Find trailing slashes */
                 if (*q != '/')
                         e = q + 1;
 
-        if (!e) /* no valid character? */
-                return -EINVAL;
+        if (!e) /* not a single non-slash character, then there is no filename to extract, return recognizable error */
+                return -EADDRNOTAVAIL;
```

Comment:
How about the following for readability?
```
if (streq(c, "/"))
    return -EADDRNOTAVAIL;

e = strchrnul(c, '/');
```

Reply:
made a similar change. ptal